ecdsa.go

```diff
@@ -188,27 +189,26 @@ func hashBytes(data []byte) []byte {
 	return buf[:]
 }
 
-// Verify verifies the signature in r, s of hash using the public key, pub. Its
-// return value records whether the signature is valid.
-func Verify(pub *ecdsa.PublicKey, hash, data []byte) error {
-	if r, s := unmarshalXY(hash); r == nil || s == nil {
-		// panic("could not unmarshal r / s")
+// Verify verifies the signature of msg using the public key pub. It returns
+// nil only if signature is valid.
+func Verify(pub *ecdsa.PublicKey, sig, msg []byte) error {
+	if r, s := unmarshalXY(sig); r == nil || s == nil {
 		return ErrCannotUnmarshal
 	} else if pub == nil {
 		return ErrEmptyPublicKey
-	} else if !ecdsa.Verify(pub, hashBytes(data), r, s) {
+	} else if !ecdsa.Verify(pub, hashBytes(msg), r, s) {
 		return errors.Wrapf(ErrInvalidSignature, "%0x : %0x", r, s)
 	}
 
 	return nil
 }
 
-// Sign signs a data using the private key. If the data is longer than
-// the bit-length of the private key's curve order, the hash will be
-// truncated to that length. It returns the signature as slice bytes.
+// Sign signs a data using the private key. If the sha56 hash of msg
```

Comment:
sha256

Reply:
we should use sha512 in future (see https://github.com/nspcc-dev/neofs-crypto/issues/4)

Reply:
Fixed